syntax = "proto3";

package entrypoint;
option go_package = "main";

message InputFeatures {
  float mean_radius = 1;
  float mean_texture = 2;
  float mean_perimeter = 3;
  float mean_area = 4;
  float mean_smoothness = 5;
  float mean_compactness = 6;
  float mean_concavity = 7;
  float mean_concave_points = 8;
  float mean_symmetry = 9;
  float mean_fractal_dimension = 10;
  float radius_error = 11;
  float texture_error = 12;
  float perimeter_error = 13;
  float area_error = 14;
  float smoothness_error = 15;
  float compactness_error = 16;
  float concavity_error = 17;
  float concave_points_error = 18;
  float symmetry_error = 19;
  float fractal_dimension_error = 20;
  float worst_radius = 21;
  float worst_texture = 22;
  float worst_perimeter = 23;
  float worst_area = 24;
  float worst_smoothness = 25;
  float worst_compactness = 26;
  float worst_concavity = 27;
  float worst_concave_points = 28;
  float worst_symmetry = 29;
  float worst_fractal_dimension = 30;
}

message Prediction {
  bool success = 1;
  string message = 2;
  bool category = 3
}

service EntryPoint {
  rpc MakePrediction(InputFeatures) returns (Prediction) {};
}